### PR TITLE
docs(gcp_pubsub source): use generated docs for config 

### LINKED ISSUE #15902: Convert the `gcp_pubsub` source to use auto-generated docs

### PR DESCRIPTION
**Pull request by StephenWakely**
Closes #15902
Epic #14998

### CODE REVIEW

File: src/sources/gcp_pubsub.rs [hunk 1]
```diff
@@ -1017,12 +1024,12 @@ mod integration_tests {
...
                 project: PROJECT.into(),
                 subscription: self.subscription.clone(),
-                endpoint: Some(gcp::PUBSUB_ADDRESS.clone()),
+                endpoint: gcp::PUBSUB_ADDRESS.clone(),
                 auth: GcpAuthConfig {
                     skip_authentication: true,
                     ..Default::default()
                 },
-                ack_deadline_secs: Some(ACK_DEADLINE as i32),
+                ack_deadline_secs: Duration::from_secs(ACK_DEADLINE),
```
Comment (spencergilbert) [inline src/sources/gcp_pubsub.rs:1032]: Could `ACK_DEADLINE` just be a `Duration`?

**Comment by spencergilbert** [comment]
This change also drops https://vector.dev/docs/reference/configuration/sources/gcp_pubsub/#gcp-pub/sub and https://vector.dev/docs/reference/configuration/sources/gcp_pubsub/#proxy

File: website/cue/reference/components/sources/gcp_pubsub.cue [hunk 2]
```diff
@@ -58,5 +58,80 @@ components: sources: gcp_pubsub: {
 		platform_name: null
 	}
 
-	configuration:  base.components.sources.gcp_pubsub.configuration
+	configuration: base.components.sources.gcp_pubsub.configuration
+
+	output: logs: record: {
```
Comment (spencergilbert) [inline website/cue/reference/components/sources/gcp_pubsub.cue:63]: Oh, haha - yeah over aggressive on the deletions there 😂

**Review by spencergilbert — APPROVED** [review]

**Comment by StephenWakely** [comment]
> This change also drops https://vector.dev/docs/reference/configuration/sources/gcp_pubsub/#gcp-pub/sub and https://vector.dev/docs/reference/configuration/sources/gcp_pubsub/#proxy

Good catch

File: src/sources/gcp_pubsub.rs [hunk 3]
```diff
@@ -144,33 +148,44 @@ pub struct PubsubConfig {
...
+    #[serde_as(as = "serde_with::DurationSeconds<f64>")]
+    pub poll_time_seconds: Duration,
 
     /// The acknowledgement deadline, in seconds, to use for this stream.
     ///
     /// Messages that are not acknowledged when this deadline expires may be retransmitted.
-    pub ack_deadline_secs: Option<i32>,
+    #[serde(default = "default_ack_deadline")]
+    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
+    pub ack_deadline_secs: Duration,
```
[the remark this answers is not in the document]
Comment (neuronull) [reply src/sources/gcp_pubsub.rs:168]: Is it worth adding a non-doc comment to that effect? (Or perhaps where the conversion is done)

**Review by neuronull — COMMENTED** [review]
Looks like some compilation errors. Otherwise LGTM

**Pull request merged**